back/app/models/user.rb

```diff
@@ -558,15 +544,27 @@ def increment_confirmation_retry_count!
     save!
   end
 
+  private
+
+  def set_confirmation_required
+    return unless new_record? && email_changed?
+
+    confirmation_not_required =
+      registered_with_phone? || sso? || invite_status.present? ||
+      active? || highest_role != :user
```

Comment:
I may be wrong but if an admin changes their email we should need them to reconfirm. Also I can't remember whether or not we allow sso users to do a lite registration ie login with just email. Be worth documenting what is and isn't possible currently: eg normally registered users can use SSO and SSO users can create passwords (or not)

Reply:
> I may be wrong but if an admin changes their email we should need them to reconfirm

Yes, but this callback is called only when a user is being created, so it's not called when a user changes their email.

> Be worth documenting what is and isn't possible currently: eg normally registered users can use SSO and SSO users can create passwords (or not)

Just added it to my draft doc here https://www.notion.so/citizenlab/Authentication-flows-51fdcbaffd54467cb363fcdb68c7620c?pvs=4#ae0fedf70c8f458fbdc7f6df95e15190